### ISSUE TITLE
Sourcemap rewriting removes DebugIds

### ISSUE THREAD

**Issue by loewenheim** [getsentry/sentry-cli#1685]
Sourcemap rewriting parses the sourcemap into a `Sourcemap` structure, modifies it, and serializes it. Since the structure doesn't know about debug ids, those get lost in the parsing. The solution is to add debug id handling to `rust-sourcemap`.
```[tasklist]
### Tasks
- [ ] https://github.com/getsentry/rust-sourcemap/pull/66
- [ ] https://github.com/getsentry/sentry-cli/pull/1686
```

**Comment by kasszz** [comment]
Hey Loewenheim, I'm currently facing an issue that kinda let me to this problem aswel, maybe you can verify it with me.

I get errors in Sentry and I have the artifact uploaded correctly. But in the Issue it states that there is no Sourcemap. But when I manually download the .js and .js.map from the artifacts that the Issue is about and run it through the ["Verify your source maps work locally"](https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/?original_referrer=https%3A%2F%2Fwww.startpage.com%2F#verify-your-source-maps-work-locally) steps. I get the correct place. 

But then I saw something about sentryDebugIds in this [Troubleshooting guide](https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/?original_referrer=https%3A%2F%2Fwww.startpage.com%2F#verify-your-source-files-contain-debug-id-injection-snippets) But in the code it's not assigned to anything:
![image](https://github.com/getsentry/sentry-cli/assets/13496198/eaedf586-d341-4eec-8f6a-832fabc883aa)

Long question short, is that what this github issue is about?

**Comment by kamilogorek** [comment]
> Long question short, is that what this github issue is about?

It's about cli removing the `//# debugId=<uuid>` pragma from the end of the file.

**Comment by loewenheim** [comment]
Fixed as of `sentry-cli` 2.20.0.
